row_group.go

```diff
@@ -277,18 +278,34 @@ func (r *rowGroupRows) buffer(i int) []Value {
 	return r.buffers[j:k:k]
 }
 
+func newRowGroupRows(rowGroup RowGroup, pageReadMode PageReadMode) *rowGroupRows {
+	return &rowGroupRows{
+		rowGroup:     rowGroup,
+		pageReadMode: pageReadMode,
+	}
+}
+
 func (r *rowGroupRows) init() {
 	columns := r.rowGroup.ColumnChunks()
 
 	r.buffers = make([]Value, len(columns)*columnBufferSize)
-	r.readers = make([]asyncPages, len(columns))
+	r.readers = make([]Pages, len(columns))
 	r.columns = make([]columnChunkRows, len(columns))
 
-	done := make(chan struct{})
-	r.done = done
-
-	for i, column := range columns {
-		r.readers[i].init(column.Pages(), done)
+	switch r.pageReadMode {
+	case PageReadModeAsync:
+		done := make(chan struct{})
+		r.done = done
+		for i, column := range columns {
+			r.readers[i] = &asyncPages{}
```

Comment:
Could we allocate all `asyncPages` in a slice so the number of malloc calls doesn't grow linearly with the number of columns?

Something like this:
```go
readers := make([]asyncPages, len(columns))
for i, column := range columns {
    readers[i].init(column.Pages(), done)
    r.readers[i] = &readers[i]
}
```

Reply:
done